unstructured/partition/pdf_image/pdfminer_processing.py

```diff
@@ -292,6 +303,38 @@ def clean_pdfminer_duplicate_image_elements(document: "DocumentLayout") -> "Docu
     return document
 
 
+@requires_dependencies("unstructured_inference")
+def remove_duplicate_embedded_text(elements: list["TextRegion"]) -> list["TextRegion"]:
+    """Removes duplicate text elements extracted by PDFMiner from a document layout."""
+    from unstructured_inference.inference.elements import EmbeddedTextRegion
+
+    bboxes = []
+    texts = []
+    bbox_to_iou_mapping = {}
```

Comment:
Wouldn't it be slightly easier if we gathered text elements and image (other) elements in two separate lists in `process_data_with_pdfminer`? We already has a "switch case" there so it would be easier and we'd avoid introducing this mapping. So it could look like:
```python
text_layout = []
image_layout = []
if hasattr(obj, "get_text"):
    # ....
    text_layout.append(text_region)
# ...
clean_text_layout = remove_duplicate_embedded_text(text_layout)
layout = [*clean_text_layout, *image_layout]
layout = sort_text_regions(layout, SORT_MODE_BASIC)
# ....
```
(ofc naming could be corrected :) )